src/ipc/ipc4/handler.c

```diff
@@ -843,11 +843,13 @@ static int ipc4_module_process_d0ix(struct ipc4_message_request *ipc4)
 		return IPC4_INVALID_RESOURCE_ID;
 	}
 
-	/* only module 0 can be used to set d0ix state */
+	/* only module 0 can be used to set d0ix state.
+	 * TODO: add secondary core support
+	 */
```

Comment:
I'm confused by what the commit message says and the code does. They're quite the opposite. And what do you mean by secondary core support? Can we really have the DSP in D0i3 with secondary cores powered up?

Reply:
"add secondary core support in future" so it is not done in this PR.  Please check IPC3 logic for D0i3, secondary core is processed. I don't add it since I am not sure whether this is applicable to MTL or not.

````c
	if (pm_gate.flags & SOF_PM_PPG) {
		pm_runtime_disable(PM_RUNTIME_DSP, PLATFORM_PRIMARY_CORE_ID);
#if (CONFIG_CAVS_LPS)
		cpu_restore_secondary_cores();
#endif
	} else {
		/* before we enable pm runtime and perform D0->D0ix flow
		 * (primary core powers off secondary cores in
		 * platform_pg_int_handler) we have to prepare all secondary
		 * cores data for powering off (disable interrupt, perform
		 * cache writeback).
		 */
#if (CONFIG_CAVS_LPS)
		cpu_secondary_cores_prepare_d0ix();
#endif
		pm_runtime_enable(PM_RUNTIME_DSP, PLATFORM_PRIMARY_CORE_ID);
	}
````

Reply:
In MTL there will not be any special case for secondary cores. This message enables dynamic PG for primary core. Later PM can decide to enter this state but will not allow it if any other core is active.

Reply:
Thanks, I will update it